classes/MusicHandler.js

```diff
@@ -8,12 +8,26 @@ module.exports = class MusicHandler {
 		this.player = player;
 	}
 
-	disconnect() {
+	async disconnect() {
 		const { bot } = require('../main.js');
+		const voiceChannel = bot.guilds.cache.get(this.player.guildId).channels.cache.get(this.player.channelId);
 		clearTimeout(this.player.timeout);
 		clearTimeout(this.player.pauseTimeout);
 		this.player.disconnect();
 		bot.music.destroyPlayer(this.player.guildId);
+		if (voiceChannel?.type === 'GUILD_STAGE_VOICE') {
+			const permissions = bot.guilds.cache.get(this.player.guildId).channels.cache.get(this.player.channelId).permissionsFor(bot.user.id);
+			if (!permissions.has(['VIEW_CHANNEL', 'CONNECT', 'SPEAK'])) return;
+			if (!permissions.has(Permissions.STAGE_MODERATOR)) return;
+			if (voiceChannel.stageInstance?.topic === getLocale(guildData.get(`${this.player.guildId}.locale`) ?? defaultLocale, 'MUSIC_STAGE_TOPIC')) {
```

Comment:
guilds can change their locale mid-stage and this check would fail

Reply:
This check is supposed return false for those cases, then it will not delete the stage instance

Reply:
Locales, i will test it.

Reply:
👍🏼